### DISCUSSION TITLE
Optimizing the Syntax Sugar of Expression Tree.

### DISCUSSION THREAD

**Discussion by NMSAzulX** [dotnet/csharplang#3927] [General] [answered]
For example:

``` C#

//old
Foo( item=>item.Name );  

//new
Foo(item.Name)
```
“Item” as the default parameter of the expression. Of course, you can also constrain it to be called ExpItem / ExpParam and so on. This constraint allows IDE to convert "item.Name" to "item= > item.Name" when parsing the syntax tree. If it is more than one parameter, it can be incremented. i.e. : 
``` C#
Foo( item1.Age == item2.Age );
```  

It's a bold idea, but I think it's useful because I've seen a lot of company code, and a lot of their code is like this: "item= > xxx", "item= > yyy", "item= > zzz". Developers spend most of their time writing "item= >".

Please let me know if I have any unclear description. :)

**Comment by DoyouNetwork** [comment]
大佬们牛逼,而我只能在翻译软件的辅助下下过来给回个帖!~~

**Reply by Unknown6656** [reply]
然后使用翻译软件将您的文字翻译成英文。;)

**Comment by WeihanLi** [comment]
Good proposal

**Comment by iceteacqx** [comment]
666

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/discussions/91
